com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/PackedResources.java

```diff
@@ -0,0 +1,163 @@
+package com.dynamo.bob.util;
+
+import com.dynamo.bob.Bob;
+import com.dynamo.bob.Platform;
+
+import java.io.File;
+import java.io.FilterInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.JarURLConnection;
+import java.net.URL;
+import java.util.jar.JarFile;
+
+public class PackedResources {
+    // Async unpack state and synchronization
+    private static volatile boolean unpackStarted = false;
+    private static volatile boolean unpackDone = false;
+    private static final Object unpackLock = new Object();
+
+    private static final Object bobInputStreamLock = new Object();
+    private static InputStream sharedBobInputStream = null;
+    private static int sharedBobInputStreamRefCount = 0;
+    private static JarFile jarFile;
+
+    private static void unpackAllLibs(Platform platform) throws IOException {
+        TimeProfiler.start("unpackAllHostLibs");
```

Comment:
Should we name the profiler entry and function the same? Perhaps `unpackHostLibs()`?

Reply:
Hmm, it's not really libs we are unpacking though, is it? `unpackHostTools()` or perhaps `unpackTools()`?

Reply:
Do we ever need to unpack files for other than the host platform?

Reply:
In general yes, we do. But not here. For example all the dmengines are in the same folders as platform tools.

I think scope names aren't that important if it's unique and redable - it's easy to understand and find in code